### PR TITLE
Add filtered 

### PR DESCRIPTION
**Pull request by safareli**
It's more powerful and I think useful, then `suchThat`, so I think it would be a nice addition.

### CODE REVIEW

File: src/Control/Monad/Gen.purs [hunk 1]
```diff
@@ -103,10 +103,17 @@ unfoldable gen = unfoldr unfold <$> sized (tailRecM loopGen <<< Tuple Nil)
 -- | matches a given predicate. This will never halt if the predicate always
 -- | fails.
 suchThat :: forall m a. MonadRec m => MonadGen m => m a -> (a -> Boolean) -> m a
-suchThat gen pred = tailRecM go unit
+suchThat gen pred = gen `mapMaybe` \a -> if pred a then Just a else Nothing
+
+-- | Creates a generator that repeatedly run another generator until its output
+-- | is Just node. This will never halt if the predicate always returns Nothing.
```
Comment (safareli) [inline src/Control/Monad/Gen.purs]: If you can suggest better wording here will be happy to change

**Comment by safareli** [comment]
I ended up using this:
```purs
filtered :: forall m a. MonadRec m => MonadGen m => m (Maybe a) -> m a
filtered gen = tailRecM go unit
  where
  go :: Unit -> m (Step Unit a)
  go _ = gen <#> \a -> case a of
    Nothing -> Loop unit
    Just a -> Done a
```

**Comment by garyb** [comment]
I think "filtered" is perhaps a better name for this too.

**Comment by safareli** [comment]
If this is fine let's `squish and merge`

**Pull request merged**